benchmark/benchmark_core/src/BM_KCE_21_SmartPool_Test.cpp

```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ *
+ *                           Klepsydra Core Modules
+ *              Copyright (C) 2019-2020  Klepsydra Technologies GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *****************************************************************************/
+#include <benchmark/benchmark.h>
+#include <klepsydra/core/cache_listener.h>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+// Considerations:
+// 1) Vector resized and published are the same vector.
+// 2) Vector resized is previously created before EventEmitterMiddlewareProvider() call
+//    with the same size which will be requested to be resized.
+// 3) Ranges.
+//    1st range: poolsize of the smartpool.
+//    2nd range: vectorSize of the vector to be resized and published.
+//    3rd range: initialization bool. true-> initialize vector. false-> do not init vector.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+static void BM_KCE_21_SmartPool_Test(benchmark::State &state)
```

Comment:
Put the right names to the tests! EventEmitter_Test

Reply:
Done.